src/Api.Rest/HttpClient/Builder/RestClientBuilder.cs

```diff
@@ -0,0 +1,346 @@
+#region copyright
+
+/* * * * * * * * * * * * * * * * * * * * * * * * * */
+/* Carl Zeiss Industrielle Messtechnik GmbH        */
+/* Softwaresystem PiWeb                            */
+/* (c) Carl Zeiss 2024                             */
+/* * * * * * * * * * * * * * * * * * * * * * * * * */
+
+#endregion
+
+namespace Zeiss.PiWeb.Api.Rest.HttpClient.Builder;
+
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Threading;
+using CacheCow.Client;
+using CacheCow.Common;
+using JetBrains.Annotations;
+using Zeiss.PiWeb.Api.Rest.Common.Authentication;
+using Zeiss.PiWeb.Api.Rest.Common.Client;
+using Zeiss.PiWeb.Api.Rest.HttpClient.Data;
+using Zeiss.PiWeb.Api.Rest.HttpClient.OAuth;
+using Zeiss.PiWeb.Api.Rest.HttpClient.RawData;
+
+/// <summary>
+/// <inheritdoc cref="Zeiss.PiWeb.Api.Rest.HttpClient.Builder.IRestClientBuilder" />
+/// Using this builder should be preferred over directly instantiating a rest client.
+/// In its default setup this builder uses an internal http cache for any rest client build. This cache is disposed when this builder is
+/// disposed. This means the builder needs to life longer then any rest client it builds. It is possible to use an external cache instead
+/// by calling <see cref="EnableExternalHttpCaching"/>.
+/// </summary>
+public class RestClientBuilder : IRestClientBuilder, IDisposable

```

Comment:
Can you please run the automatic code cleanup over this file, it is missing the regions, for example. As far as I can see this is the only file where it is missing.